File: php/src/app/dominio/Contexto.php
<?php
namespace dominio;

use dominio\estrategias\Estrategia;

class Contexto {
    /**
     * @var Estrategia
     */
    private $estrategia;

    public function actuar(){
        $this->estrategia->ejecutar();
    }

    public function cambiarEstrategia(Estrategia $estrategia){
        $this->estrategia = $estrategia;
    }
}

File: java/src/main/java/dominio/Contexto.java
package dominio;

import dominio.estrategias.Estrategia;

public class Contexto {
    private Estrategia estrategia;

    public void actuar(){
        this.estrategia.ejecutar();
    }

    public void cambiarEstrategia(Estrategia estrategia){
        this.estrategia = estrategia;
    }
}